File: string-paradigm-expression_1/src/main/java/com/steelzack/string/paradigm1/expression/RegionImpl.java
package com.steelzack.string.paradigm1.expression;

import com.steelzack.string.paradigm.api.Region;

public class RegionImpl extends Region {
	public RegionImpl(String type, String size) {
		this.regionName = type;
		this.assignedNumber = size;
	}

	public String getRegionName() {
		return regionName;
	}

	public String getAssignedNumber() {
		return assignedNumber;
	}

	public String getDescription() {
		return regionName + ", " + assignedNumber;
	}

	@Override
	public String toString() {
		return getDescription();
	}
}

File: string-paradigm-expression_1/src/main/kotlin/org/jesperancinha/string/paradigm1/expression/RegionImpl.kt
package org.jesperancinha.string.paradigm1.expression

import org.jesperancinha.string.paradigm.api.Region

class RegionImpl(type: String, size: String) : Region(type, size) {
    override var description: String
        get() = "$regionName, $assignedNumber"
        set(description) {
            super.description = description
        }
    override fun toString(): String {
        return description
    }
}